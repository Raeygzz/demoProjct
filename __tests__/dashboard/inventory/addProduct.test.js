import 'react-native';
import React from 'react';
import AddProduct from '../../../src/screens/dashboard/inventory/addProduct';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<AddProduct />).toJSON();
  expect(tree).toMatchSnapshot();
});
