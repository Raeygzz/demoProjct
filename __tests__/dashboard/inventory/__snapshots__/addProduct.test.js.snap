// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  transparent={false}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "paddingHorizontal": 27,
          "paddingVertical": 50,
        },
        Object {
          "borderTopLeftRadius": 24,
          "borderTopRightRadius": 24,
        },
        Object {
          "backgroundColor": "#171635",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#10354b",
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": -1,
          },
          undefined,
        ]
      }
    >
      <BVLinearGradient
        angle={135}
        colors={
          Array [
            4288230089,
            4283570823,
            4279121480,
            4280039002,
          ]
        }
        endPoint={
          Object {
            "x": 1,
            "y": 1,
          }
        }
        locations={
          Array [
            0,
            0.36087942,
            0.8752376,
            1,
          ]
        }
        startPoint={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        style={
          Object {
            "flex": 1,
            "height": 1534,
            "opacity": 0.32,
            "position": "relative",
            "zIndex": -1,
          }
        }
        useAngle={true}
      />
    </View>
    <RCTScrollView
      keyboardShouldPersistTaps="handled"
      style={
        Object {
          "height": "100%",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "height": "100%",
            }
          }
        >
          <Text
            minimumFontScale={16}
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Apercu_Regular",
                  "fontSize": 16,
                },
                Object {
                  "fontFamily": "Apercu-Bold",
                  "fontSize": 14,
                  "lineHeight": 17,
                  "marginBottom": 30,
                  "opacity": 0.48,
                },
              ]
            }
          >
            ADD PRODUCTS
          </Text>
          <View
            style={
              Array [
                Object {
                  "display": "flex",
                  "justifyContent": "flex-end",
                  "marginBottom": 20,
                  "minHeight": 68,
                  "overflow": "hidden",
                  "paddingTop": 20,
                  "position": "relative",
                },
                undefined,
                Object {
                  "paddingBottom": 0,
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontFamily": "Apercu_Regular",
                  "fontSize": 14,
                  "left": 0,
                  "lineHeight": 17,
                  "opacity": 0.68,
                  "position": "absolute",
                  "top": 26,
                }
              }
            >
              Product Name
            </Text>
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              autoFocus={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onSubmitEditing={null}
              placeholderTextColor="rgba(255,255,255,0.7)"
              rejectResponderTermination={true}
              returnKeyType="done"
              selectionColor="#71648E"
              style={
                Array [
                  Object {
                    "borderBottomColor": "white",
                    "borderBottomWidth": 1,
                    "color": "white",
                    "fontFamily": "Apercu Medium",
                    "fontSize": 18,
                    "padding": 0,
                  },
                  null,
                  undefined,
                  Object {
                    "borderBottomColor": "white",
                    "height": 37,
                    "lineHeight": 12,
                  },
                  null,
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "borderBottomColor": "white",
                  "borderBottomWidth": 1,
                  "height": 68,
                },
                Object {
                  "display": "flex",
                  "justifyContent": "flex-end",
                  "marginBottom": 20,
                  "minHeight": 68,
                  "overflow": "hidden",
                  "paddingTop": 20,
                  "position": "relative",
                },
                Object {
                  "flex": 1,
                },
              ]
            }
          >
            <View>
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#71648E",
                      "fontSize": 12,
                    },
                    Object {
                      "position": "absolute",
                      "right": 18,
                      "top": 18,
                    },
                    Object {
                      "fontFamily": "simple-line-icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                î˜„
              </Text>
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "Apercu_Regular",
                    "fontSize": 14,
                    "left": 0,
                    "lineHeight": 17,
                    "opacity": 0.68,
                    "position": "absolute",
                    "top": 16,
                  }
                }
              >
                Product Quantity
              </Text>
              <Text
                minimumFontScale={16}
                style={
                  Array [
                    Object {
                      "color": "#fff",
                      "fontFamily": "Apercu_Regular",
                      "fontSize": 16,
                    },
                    Array [
                      Object {
                        "bottom": 16,
                        "fontSize": 18,
                        "left": 0,
                        "position": "absolute",
                      },
                      Object {
                        "fontFamily": "Apercu-Medium",
                      },
                    ],
                  ]
                }
              >
                
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "height": 50,
                      "margin": 0,
                      "marginLeft": -9,
                      "opacity": 0,
                      "padding": 0,
                    },
                    Object {
                      "fontFamily": "Apercu-Bold",
                    },
                    Object {
                      "borderBottomColor": "white",
                      "borderBottomWidth": 1,
                      "color": "white",
                      "fontFamily": "Apercu Medium",
                      "fontSize": 18,
                      "padding": 0,
                    },
                    null,
                    Object {
                      "color": "white",
                    },
                  ]
                }
              >
                <RCTPicker
                  items={
                    Array [
                      Object {
                        "label": "Select Product Quantity",
                        "textColor": undefined,
                        "value": null,
                      },
                      Object {
                        "label": "01",
                        "textColor": undefined,
                        "value": "01",
                      },
                      Object {
                        "label": "02",
                        "textColor": undefined,
                        "value": "02",
                      },
                      Object {
                        "label": "03",
                        "textColor": undefined,
                        "value": "03",
                      },
                      Object {
                        "label": "04",
                        "textColor": undefined,
                        "value": "04",
                      },
                      Object {
                        "label": "05",
                        "textColor": undefined,
                        "value": "05",
                      },
                      Object {
                        "label": "06",
                        "textColor": undefined,
                        "value": "06",
                      },
                      Object {
                        "label": "07",
                        "textColor": undefined,
                        "value": "07",
                      },
                      Object {
                        "label": "08",
                        "textColor": undefined,
                        "value": "08",
                      },
                      Object {
                        "label": "09",
                        "textColor": undefined,
                        "value": "09",
                      },
                      Object {
                        "label": "10",
                        "textColor": undefined,
                        "value": "10",
                      },
                      Object {
                        "label": "11",
                        "textColor": undefined,
                        "value": "11",
                      },
                      Object {
                        "label": "12",
                        "textColor": undefined,
                        "value": "12",
                      },
                      Object {
                        "label": "13",
                        "textColor": undefined,
                        "value": "13",
                      },
                      Object {
                        "label": "14",
                        "textColor": undefined,
                        "value": "14",
                      },
                      Object {
                        "label": "15",
                        "textColor": undefined,
                        "value": "15",
                      },
                      Object {
                        "label": "16",
                        "textColor": undefined,
                        "value": "16",
                      },
                      Object {
                        "label": "17",
                        "textColor": undefined,
                        "value": "17",
                      },
                      Object {
                        "label": "18",
                        "textColor": undefined,
                        "value": "18",
                      },
                      Object {
                        "label": "19",
                        "textColor": undefined,
                        "value": "19",
                      },
                      Object {
                        "label": "20",
                        "textColor": undefined,
                        "value": "20",
                      },
                    ]
                  }
                  onChange={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  selectedIndex={0}
                  style={
                    Array [
                      Object {
                        "height": 216,
                      },
                      undefined,
                    ]
                  }
                />
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "display": "flex",
                  "justifyContent": "flex-end",
                  "marginBottom": 20,
                  "minHeight": 68,
                  "overflow": "hidden",
                  "paddingTop": 20,
                  "position": "relative",
                },
                undefined,
                Object {
                  "paddingBottom": 0,
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontFamily": "Apercu_Regular",
                  "fontSize": 14,
                  "left": 0,
                  "lineHeight": 17,
                  "opacity": 0.68,
                  "position": "absolute",
                  "top": 26,
                }
              }
            >
              Product Per Amount
            </Text>
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onSubmitEditing={[Function]}
              placeholderTextColor="rgba(255,255,255,0.7)"
              rejectResponderTermination={true}
              returnKeyType="next"
              selectionColor="#71648E"
              style={
                Array [
                  Object {
                    "borderBottomColor": "white",
                    "borderBottomWidth": 1,
                    "color": "white",
                    "fontFamily": "Apercu Medium",
                    "fontSize": 18,
                    "padding": 0,
                  },
                  null,
                  undefined,
                  Object {
                    "borderBottomColor": "white",
                    "height": 37,
                    "lineHeight": 12,
                  },
                  null,
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "display": "flex",
                  "justifyContent": "flex-end",
                  "marginBottom": 20,
                  "minHeight": 68,
                  "overflow": "hidden",
                  "paddingTop": 20,
                  "position": "relative",
                },
                undefined,
                Object {
                  "paddingBottom": 0,
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontFamily": "Apercu_Regular",
                  "fontSize": 14,
                  "left": 0,
                  "lineHeight": 17,
                  "opacity": 1,
                  "position": "absolute",
                  "top": 20,
                }
              }
            >
              Product Total Amount
            </Text>
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              editable={false}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onSubmitEditing={null}
              placeholderTextColor="rgba(255,255,255,0.7)"
              rejectResponderTermination={true}
              returnKeyType="done"
              selectionColor="#71648E"
              style={
                Array [
                  Object {
                    "borderBottomColor": "white",
                    "borderBottomWidth": 1,
                    "color": "white",
                    "fontFamily": "Apercu Medium",
                    "fontSize": 18,
                    "padding": 0,
                  },
                  null,
                  undefined,
                  Object {
                    "borderBottomColor": "white",
                    "height": 37,
                    "lineHeight": 12,
                  },
                  null,
                ]
              }
              underlineColorAndroid="transparent"
              value="1000"
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "display": "flex",
                  "justifyContent": "flex-end",
                  "marginBottom": 20,
                  "minHeight": 68,
                  "overflow": "hidden",
                  "paddingTop": 20,
                  "position": "relative",
                },
                undefined,
                Object {
                  "paddingBottom": 0,
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontFamily": "Apercu_Regular",
                  "fontSize": 14,
                  "left": 0,
                  "lineHeight": 17,
                  "opacity": 0.68,
                  "position": "absolute",
                  "top": 26,
                }
              }
            >
              Product Description
            </Text>
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              multiline={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onSubmitEditing={null}
              placeholderTextColor="rgba(255,255,255,0.7)"
              rejectResponderTermination={true}
              returnKeyType="done"
              selectionColor="#71648E"
              style={
                Array [
                  Object {
                    "borderBottomColor": "white",
                    "borderBottomWidth": 1,
                    "color": "white",
                    "fontFamily": "Apercu Medium",
                    "fontSize": 18,
                    "padding": 0,
                  },
                  null,
                  undefined,
                  Object {
                    "borderBottomColor": "white",
                    "height": 37,
                    "lineHeight": 12,
                  },
                  null,
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
          <View
            style={
              Object {
                "height": 20,
              }
            }
          >
            <Text>
              Â 
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              minimumFontScale={16}
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontFamily": "Apercu_Regular",
                    "fontSize": 16,
                  },
                  Object {
                    "color": "yellow",
                  },
                ]
              }
            >
              Upload Image
            </Text>
          </View>
          <View
            style={
              Object {
                "height": 20,
              }
            }
          >
            <Text>
              Â 
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderBottomColor": "white",
                  "borderBottomWidth": 1,
                  "height": 68,
                },
                Object {
                  "display": "flex",
                  "justifyContent": "flex-end",
                  "marginBottom": 20,
                  "minHeight": 68,
                  "overflow": "hidden",
                  "paddingTop": 20,
                  "position": "relative",
                },
                Object {
                  "flex": 1,
                },
              ]
            }
          >
            <View>
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#71648E",
                      "fontSize": 12,
                    },
                    Object {
                      "position": "absolute",
                      "right": 18,
                      "top": 18,
                    },
                    Object {
                      "fontFamily": "simple-line-icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                î˜„
              </Text>
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "Apercu_Regular",
                    "fontSize": 14,
                    "left": 0,
                    "lineHeight": 17,
                    "opacity": 0.68,
                    "position": "absolute",
                    "top": 16,
                  }
                }
              >
                Product Market scale
              </Text>
              <Text
                minimumFontScale={16}
                style={
                  Array [
                    Object {
                      "color": "#fff",
                      "fontFamily": "Apercu_Regular",
                      "fontSize": 16,
                    },
                    Array [
                      Object {
                        "bottom": 16,
                        "fontSize": 18,
                        "left": 0,
                        "position": "absolute",
                      },
                      Object {
                        "fontFamily": "Apercu-Medium",
                      },
                    ],
                  ]
                }
              >
                
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "height": 50,
                      "margin": 0,
                      "marginLeft": -9,
                      "opacity": 0,
                      "padding": 0,
                    },
                    Object {
                      "fontFamily": "Apercu-Bold",
                    },
                    Object {
                      "borderBottomColor": "white",
                      "borderBottomWidth": 1,
                      "color": "white",
                      "fontFamily": "Apercu Medium",
                      "fontSize": 18,
                      "padding": 0,
                    },
                    null,
                    Object {
                      "color": "white",
                    },
                  ]
                }
              >
                <RCTPicker
                  items={
                    Array [
                      Object {
                        "label": "Select Product Market Scale",
                        "textColor": undefined,
                        "value": null,
                      },
                      Object {
                        "label": "High",
                        "textColor": undefined,
                        "value": "High",
                      },
                      Object {
                        "label": "Moderate",
                        "textColor": undefined,
                        "value": "Moderate",
                      },
                      Object {
                        "label": "Low",
                        "textColor": undefined,
                        "value": "Low",
                      },
                      Object {
                        "label": "Fresher",
                        "textColor": undefined,
                        "value": "Fresher",
                      },
                      Object {
                        "label": "Deprecated",
                        "textColor": undefined,
                        "value": "Deprecated",
                      },
                    ]
                  }
                  onChange={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  selectedIndex={0}
                  style={
                    Array [
                      Object {
                        "height": 216,
                      },
                      undefined,
                    ]
                  }
                />
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-around",
                "marginVertical": 30,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#ffbcbc",
                    "borderColor": "red",
                    "borderWidth": 1,
                    "height": 100,
                    "justifyContent": "center",
                    "margin": 10,
                    "width": 300,
                  },
                  Object {
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                style={
                  Object {
                    "color": "#333333",
                    "margin": 20,
                  }
                }
              >
                TouchableNativeFeedback is not supported on this platform!
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#ffbcbc",
                    "borderColor": "red",
                    "borderWidth": 1,
                    "height": 100,
                    "justifyContent": "center",
                    "margin": 10,
                    "width": 300,
                  },
                  Object {
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                style={
                  Object {
                    "color": "#333333",
                    "margin": 20,
                  }
                }
              >
                TouchableNativeFeedback is not supported on this platform!
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</Modal>
`;
